### ISSUE TITLE
Quick Tags (and Trim Reblogs and Quick Flags PR): tipping status lost

### ISSUE THREAD

**Issue by marcustyphoon** [AprilSylph/XKit-Rewritten#838]
### Platform

MacOS 10.15.7 x86

### Browser

Ungoogled Chromium v107.0.5304.110

### Addon version

v0.21.0

### Details

Adding a tag to an original post with Quick Tags turns off tipping on the post; we don't pass through `canBeTipped` and I guess undefined is falsey.

This presumably technically isn't an issue in Trim Reblogs, since you can't have tipping on a non-original post and you can only run Trim Reblogs on those, but more generally, an audit of passthrough fields that may be missing from our post-editing code seems wise.

> For editing a post, all of the request parameters from the NPF Post Creation route are expected (depending on if it's an original post or reblog), along with the Post's ID in the query path. If you are editing a scheduled post, make sure to include its publish_on value.
> https://github.com/tumblr/docs/blob/master/api.md#postspost-id---editing-a-post-neue-post-format

can_be_tipped
community_label_categories
has_community_label
hide_trail
interactability_reblog
placement_id (and possibly earned_id?)
send_to_twitter

#827 #839

**Comment by marcustyphoon** [comment]
proposal:
  
```js

import { filterPropsForEditing } from './util/something.js'

const {
    response: {
      tags = [],
      ...rest
    }
  } = await apiFetch(`/v2/blog/${uuid}/posts/${postId}`);

  tags.push(...tagsToAdd);

  try {
    const { response: { displayText } } = await apiFetch(`/v2/blog/${uuid}/posts/${postId}`, {
      method: 'PUT',
      body: {
        ...filterPropsForEditing(rest),
        tags: tags.join(','),
      }
    });

**Comment by marcustyphoon** [comment]
gutenblr:

```js
i()(this, "getPostData", (()=>{
  const {editablePost: e, editorValue: t, tags: n} = this.state
    , {trail: r} = this.props
    , s = this.blockEditorRef.current?.getContent()
    , o = s?.content ?? (0,
  Yt.uM)(t)
    , i = o.findIndex(we.KM)
    , a = i > -1;
  a && o.splice(i, 1);
  const l = s?.layout ?? e?.layout?.filter(we.af) ?? [];
  if (!s) {
      const e = (0,
      Yt.lZ)(t)
        , n = e.display.some((e=>e.blocks.length > 1));
      (a || n) && (a && (e.truncateAfter = i - 1),
      l.push(e))
  }
  const u = {
      content: o,
      tags: n,
      ...l.length && {
          layout: l
      },
      hideTrail: !this.state.showFullTrail,
      id: e?.id,
      sourceUrl: e?.sourceUrl,
      slug: e?.slug,
      date: e?.date,
      placementId: e?.placementId,
      earnedId: e?.earnedId,
      canBeTipped: e?.canBeTipped ?? null,
      interactabilityReblog: e?.interactabilityReblog
  };
  u.sendToTwitter = !1;
  const c = r?.length ? r[r.length - 1] : void 0;
  return !u.reblogKey && (0,
  we.eb)(c) && c.post.reblogKey && (u.reblogKey = c.post.reblogKey,
  u.parentTumblelogUuid = c.blog.uuid,
  u.parentPostId = c.post.id,
  u.interactabilityReblog = c.post.interactabilityReblog),
  u
}
```

apiHelper.postAPost:

```js
const $ = (0,
h.v)((0,
w.c)({
    reblogType: a,
    placementId: i,
    layout: B,
    state: A,
    sendToTwitter: T,
    adsParams: S,
    reblogKey: x,
    hideTrail: _,
    parentTumblelogUuid: R,
    parentPostId: f,
    sourceUrl: I,
    publishOn: N,
    content: v,
    slug: M,
    date: L,
    isPrivate: F,
    earnedId: E,
    canBeTipped: D,
    interactabilityReblog: z,
    tags: k?.join(", "),
    hasCommunityLabel: G,
    communityLabelCategories: W
}, ((e,t)=>void 0 !== t)));
return new U(e,`/v2/blog/${t}/posts${H ? `/${y}` : ""}`,{
    body: $,
    method: H ? "PUT" : "POST"
})
```

**Comment by marcustyphoon** [comment]
Fixed by #879, though I still think this should be fixed in Tumblr's back end.
